resources/bolt_bash_completion.sh

```diff
@@ -6,134 +6,195 @@
+      module)
+        if [ $COMP_CWORD -eq 2 ]; then
+          next="add generate-types install show"
+        else
+          case $action in
+            add | install) next="--project";;
+            generate-types) next="${context_opts}";;
+            show) next="${context_opts} --filter --format";;
+          esac
+        fi
+        ;;
+      plan)
+        if [ $COMP_CWORD -eq 2 ]; then
+          next="convert new run show"
+        elif [[ ($COMP_CWORD -eq 3 || \
+          ( $COMP_CWORD -eq 4 && "$cur" == *"::" ) || \
+          ( $COMP_CWORD -eq 5 && "$cur" == *"::"* )) &&
+          $action != "new" ]]; then
 
-	case $prev in
-	*bolt)
-		next="command file task plan project group inventory secret script apply"
-		;;
-	command | script)
-		next="run"
-		;;
-	task)
-		next="show run"
-		;;
-	plan)
-		next="show run convert"
-		;;
-	file)
-		next="upload"
-		;;
-	group | inventory)
-		next="show"
-		;;
-	secret)
-		next="createkeys encrypt decrypt"
-		;;
-	encrypt | decrypt | createkeys)
-		next="-m --modulepath --plugin ${project_config_opts}"
-		[[ $prev == "createkeys" ]] && next="${next} --force"
-		;;
-	module)
-		next="install show generate-types"
-		;;
-	install)
-		next="--log-level -m --modulepath ${project_config_opts}"
-		;;
-	show-modules | generate-types)
-		next="--log-level -m --modulepath ${project_config_opts}"
-		;;
-	convert)
-		next="--log-level -m -modulepath ${project_config_opts}"
-		;;
-	show)
-		if [ "$prevprev" == "group" ]; then
-			next="${project_config_opts} -i --inventoryfile --log-level"
-		elif [ "$prevprev" == "inventory" ]; then
-			next="${targeting_opts} ${project_config_opts} --detail -i --inventoryfile --log-level"
-		elif [ "$prevprev" == "plan" ] || [ "$prevprev" == "task" ]; then
-			next="${general_opts} ${project_config_opts} -m --modulepath --filter --format"
-		fi
-		;;
-	run)
-		next="${all_options} ${targeting_opts} --params --tmpdir --ssh-command --copy-command"
-		[[ ${prevprev} == "plan" ]] && next="${next} --compile-concurrency --hiera-config"
-		;;
-	upload | apply)
-		next="${all_options} ${targeting_opts} --ssh-command --copy-command"
-		[[ ${prev} == "apply" ]] && next="${next} --compile-concurrency --hiera-config"
-		;;
-	project)
-		next="init migrate"
-		;;
-	init)
-		next="--log-level --modules"
-		;;
-	migrate)
-		next="--log-level -i --inventoryfile ${project_config_opts}"
-		;;
-	-t | --targets)
-		if [[ -f ${inventory_list_cache_file} ]]; then
-			local next=$(cat ${inventory_list_cache_file})
-		else
-			local jsondata=$(bolt --detail --targets all inventory show | sed -e '$d')
-			jsondata=${jsondata//\"/\\\"}
+          if [ -f "${PWD}/.plan_cache.json" ]; then
+            # Use Puppet's ruby instead of jq since we know it will be there.
+            next=$(get_json_keys "${PWD}/.plan_cache.json")
+          elif [ -f "${HOME}/.puppetlabs/bolt/.plan_cache.json" ]; then
+            next=$(get_json_keys "${HOME}/.puppetlabs/bolt/.plan_cache.json")
+          fi
```

Comment:
I think this also needs the same `else` statement as the `task` command so the options show if `.plan_cache.json` doesn't exist.

Reply:
Good catch!